terraform/ecs/linux/ec2_launch/daemon/main.tf

```diff
@@ -0,0 +1,228 @@
+    testing_id                   = random_id.testing_id.hex
+  }
+}
+
+resource "aws_ecs_task_definition" "cwagent_task_definition" {
+  family                   = "cwagent-task-family-${random_id.testing_id.hex}"
+  network_mode             = "bridge"
+  task_role_arn            = aws_iam_role.ecs_task_role.arn
+  execution_role_arn       = aws_iam_role.ecs_task_execution_role.arn
+  cpu                      = 256
+  memory                   = 2048
+  requires_compatibilities = ["EC2"]
+  container_definitions    = data.template_file.cwagent_container_definitions.rendered
+  depends_on               = [aws_cloudwatch_log_group.log_group, aws_iam_role.ecs_task_role, aws_iam_role.ecs_task_execution_role]
+  volume{
+    name= "proc"
+    host_path= "/proc"
+  }
+  volume{
+    name= "dev"
+    host_path= "/dev"
+  }
+  volume{
+    name= "al1_cgroup"
+    host_path= "/cgroup"
+  }
+  volume{
+    name= "al2_cgroup"
+    host_path= "/sys/fs/cgroup"
+  }
+}
+
+resource "aws_ecs_service" "cwagent_service" {
+  name            = "cwagent-service-${random_id.testing_id.hex}"
+  cluster         = aws_ecs_cluster.cluster.id
+  task_definition = aws_ecs_task_definition.cwagent_task_definition.arn
+  launch_type     = "EC2"
+  scheduling_strategy = "DAEMON"
+
+  depends_on = [aws_iam_role_policy_attachment.ecs_task_execution_role]
+}
+
+#####################################################################
+# Sample app for scrapping metrics and logs and sending to cloudwatch
+#####################################################################
+
+data "template_file" "extra_apps" {
+  template = file(local.extra_apps_ecs_taskdef)
+  vars = {
+    region    = var.region
+    log_group = aws_cloudwatch_log_group.log_group.name
+  }
+}
+
+resource "aws_ecs_task_definition" "extra_apps_task_definition" {
+  family                   = "extra-apps-family-${random_id.testing_id.hex}"
+  network_mode             = "awsvpc"
+  task_role_arn            = aws_iam_role.ecs_task_role.arn
+  execution_role_arn       = aws_iam_role.ecs_task_execution_role.arn
+  cpu                      = 256
+  memory                   = 1024
+  requires_compatibilities = ["FARGATE"]
+  container_definitions    = data.template_file.extra_apps.rendered
+  depends_on               = [aws_cloudwatch_log_group.log_group, aws_iam_role.ecs_task_role, aws_iam_role.ecs_task_execution_role]
+}
+
+resource "aws_ecs_service" "extra_apps_service" {
+  name            = "extra-apps-service-${random_id.testing_id.hex}"
+  cluster         = aws_ecs_cluster.cluster.id
+  task_definition = aws_ecs_task_definition.extra_apps_task_definition.arn
+  desired_count   = 1
+  launch_type     = "FARGATE"
+
+  network_configuration {
+    security_groups  = [aws_security_group.ecs_security_group.id]
+    subnets          = toset(data.aws_subnets.default.ids)
+    assign_public_ip = true
+  }
+
+  depends_on = [aws_iam_role_policy_attachment.ecs_task_execution_role]
+}
+
+resource "null_resource" "validator" {
```

Comment:
We can change to `ssh` into EC2 instead of `local-exe` that similar to other EC2 test. Since there are no use case that shares between EC2 and Fargate (except Prometheus)

Reply:
what's the benefit? also not extensible if there's a test case with multiple ec2 instances behind the same ecs cluster.